Source/CombatExtended/CombatExtended/Comps/CompMeleeTargettingGizmo.cs

```diff
@@ -0,0 +1,254 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Verse;
+using RimWorld;
+using UnityEngine;
+
+namespace CombatExtended
+{
+    [StaticConstructorOnStartup]
+    public class MeleeTargettingAdd
+    {
+        static MeleeTargettingAdd()
+        {
+            foreach(ThingDef humanlike in DefDatabase<ThingDef>.AllDefs.Where(y => y.race != null && y.race.Humanlike))
+            {
+                if (humanlike.comps == null)
+                {
+                    humanlike.comps = new List<CompProperties>();
+                }
+
+                humanlike.comps.Add(new CompProperties { compClass = typeof(CompMeleeTargettingGizmo) });
+            }
+        }
+    }
+    public class CompMeleeTargettingGizmo : ThingComp
+    {
+        #region Fields
+        public Pawn PawnParent => (Pawn)this.parent;
+
+        public BodyPartHeight heightInt = BodyPartHeight.Undefined;
+
+        public BodyPartDef targetBodyPart = BodyPartDefOf.Torso;
+
+        #endregion
+
+        #region Properties
+
+        public Thing primaryWeapon
+        {
+            get
+            {
+                Thing result = null;
+
+
+                result = PawnParent?.equipment.Primary;
+
+                return result;
+            }
+        }
+
+        public BodyPartHeight finalHeight(Pawn target)
+        {
+            if (PawnParent.Faction == Faction.OfPlayer && heightInt != BodyPartHeight.Undefined)
+            {
+                return heightInt;
+            }
+
+            float maxWeaponPen = 1f;
+
+            if (primaryWeapon != null)
+            {
+                maxWeaponPen = primaryWeapon.def.tools.Select(x => x as ToolCE).Max(x => x.armorPenetrationSharp) * primaryWeapon.GetStatValue(CE_StatDefOf.MeleePenetrationFactor);
+            }
+
+            if (PawnParent.skills.GetSkill(SkillDefOf.Melee).Level < 16 && PawnParent.skills.GetSkill(SkillDefOf.Melee).Level > 4)

```

Comment:
When level == 16, this block will be skipped, and this will also be skipped.  And the subsequent block will be skipped.  This should probably be < 16, and the following should be > 15.